src/ethereum_test_tools/tests/test_filling/test_fixtures.py

```diff
@@ -173,311 +175,377 @@ def test_fill_state_test(
+    ):
+        t8n = GethTransitionTool()
+        return BlockchainTest(
+            pre=pre,
+            post=post,
+            blocks=blocks,
+            genesis_environment=genesis_environment,
+            tag="my_blockchain_test_valid_txs",
+            fixture_format=fixture_format,
+        ).generate(
+            t8n=t8n,
+            fork=fork,
+        )
 
-    if solc_version >= SOLC_PADDING_VERSION:
-        expected = expected["solc=padding_version"]
-    else:
-        expected = expected[f"solc={solc_version}"]
+    @pytest.mark.parametrize("fork", [London, Shanghai], indirect=True)
+    def test_fill_blockchain_valid_txs(  # noqa: D102
+        self,
+        fork: Fork,
+        solc_version: str,
+        check_hive: bool,
+        fixture_format: FixtureFormats,
+        expected_json_file: str,
+        blockchain_test_fixture: BlockchainFixture,
+    ):
+        assert blockchain_test_fixture.format() == fixture_format
+        assert isinstance(blockchain_test_fixture, BlockchainFixtureCommon)
+
+        fixture = {
+            f"000/my_blockchain_test/{fork.name()}": blockchain_test_fixture.to_json(),
+        }
+
+        with open(
+            os.path.join(
+                "src",
+                "ethereum_test_tools",
+                "tests",
+                "test_filling",
+                "fixtures",
+                expected_json_file,
+            )
+        ) as f:
+            expected = json.load(f)
+
+        fixture_json = to_json(fixture)
+        remove_info(fixture_json)
+
+        if solc_version >= SOLC_PADDING_VERSION:
+            expected = expected["solc=padding_version"]
+        else:
+            expected = expected[f"solc={solc_version}"]
+
+        assert fixture_json == expected
+
+    @pytest.mark.parametrize("fork", [London], indirect=True)
+    def test_fixture_header_join(self, blockchain_test_fixture: BlockchainFixture):
+        """
+        Test `FixtureHeader.join()`.
+        """
+        block = blockchain_test_fixture.blocks[0]
+        new_difficulty = block.block_header.difficulty - 1  # type: ignore
+
+        new_state_root = Hash(12345)
+        # See description of https://github.com/ethereum/execution-spec-tests/pull/398
+        new_transactions_root = "0x100"
+        header_new_fields = Header(
+            difficulty=new_difficulty,
+            state_root=new_state_root,
+            transactions_root=new_transactions_root,
+        )
 
-    assert fixture_json == expected
+        updated_block_header = block.block_header.join(header_new_fields)  # type: ignore
+        assert updated_block_header.difficulty == new_difficulty
+        assert updated_block_header.state_root == new_state_root
+        assert updated_block_header.transactions_root == Hash(new_transactions_root)
+        assert updated_block_header.hash == block.block_header.hash  # type: ignore
+        assert isinstance(updated_block_header.transactions_root, Hash)
```

Comment:
I didn't manage to resolve these mypy errors, so disabled them (`# type: ignore`). Feel free to fix them, if you know how.
```
 ➜ mypy src setup.py
src/ethereum_test_tools/tests/test_filling/test_fixtures.py:532: error: Item "InvalidFixtureBlock" of "Union[FixtureBlock, InvalidFixtureBlock]" has no attribute "block_header"
src/ethereum_test_tools/tests/test_filling/test_fixtures.py:542: error: Item "InvalidFixtureBlock" of "Union[FixtureBlock, InvalidFixtureBlock]" has no attribute "block_header"
src/ethereum_test_tools/tests/test_filling/test_fixtures.py:548: error: Item "InvalidFixtureBlock" of "Union[FixtureBlock, InvalidFixtureBlock]" has no attribute "block_header"
```

Reply:
I think it's ok, the other option is to add a bunch of assertions to check the types, but it's redundant.